### ISSUE TITLE
Need the blur effect to work on alpha channels.

### ISSUE THREAD

**Issue by baterax** [Xaymar/obs-StreamFX#814]
### Explain the Feature/Enhancement

I've been using blur on mirrored sources, it's beautiful for a frosted glass effect.
However, it only works if I'm not greenscreening.
If I want to have it on a source or scene that I have green screen on, the edges don't blur, so it ruins the effect.
From my understanding and some advice from a friend, this means the blur isn't happening on the alpha channel.
Should be an easy change.

### What makes this feature/enhancement useful for StreamFX and its users?

This will provide more versatility as people will be able to use the blur when they green screens. Attached are two files that show what I mean.
![Screenshot 2022-06-05 20-14-31](https://user-images.githubusercontent.com/8533902/172067227-927d6357-84c4-4570-8b1f-62f80159e7e0.png)
![Screenshot 2022-06-05 20-13-40](https://user-images.githubusercontent.com/8533902/172067228-d845dbc5-0acd-4e78-90c5-fe532b2a3da8.png)

### Checklist

- [ ] I am offering a monetary bounty/reward for the implementation of this Feature/Enhancement.
- [X] There is no other way to currently do this in StreamFX or OBS Studio.

**Comment by Xaymar** [comment]
The images do not show what you mean.

**Comment by Xaymar** [comment]
Did you want to request Channel Blur? Channel Blur is different from normal Blur.

**Comment by baterax** [comment]
The photo does show what I mean, if you look at the one with the blue/purple background, you can see my elbow is not blurred, you can see the sharp edge.

**Comment by baterax** [comment]
It's also visible on my finger below the game scene on the right side.

**Comment by skeletonbow** [comment]
If I am following this correctly, you are chroma keyed on top of a substituted background, and are trying to blur your arm in the lower third area.  Blur affects the color data but as far as I'm aware does not affect the alpha channel, so if the image being blurred contains transparent pixels their transparency will not be altered and the blur should not spill over into what is transparent in the source image.

If I'm understanding your request, you're asking for an option to have otherwise transparent pixels on the edges of an image become opaque so that the opaque pixels within the image can blur outside of the source image boundary into the transparent areas?

**Comment by skeletonbow** [comment]
Is this what you're trying to do?
![Screenshot 2022-06-06 14-36-55](https://user-images.githubusercontent.com/49483514/172224660-98d5feb9-a61b-4f6d-ad16-4cc479e5fe6d.png)

**Comment by skeletonbow** [comment]
![Screenshot 2022-06-06 14-40-26](https://user-images.githubusercontent.com/49483514/172225405-a8fae989-fb3f-4848-834d-77fa73ea4fc7.png)

**Comment by baterax** [comment]
YES. Awesome, I'll test!

**Comment by baterax** [comment]
So this is that setup with Box Linear.
It just doesn't want to do it.
![Screenshot 2022-06-06 20-50-17](https://user-images.githubusercontent.com/8533902/172236993-68025e9d-51d1-4c33-be37-66344f494938.png)

**Comment by Xaymar** [comment]
Blur is for Color only, not Alpha. Applying it to Alpha is what Channel Blur would be for.

**Comment by baterax** [comment]
So there IS a channel blur effect!?
That would've been easier lol.
Thank you, I'll try that.

**Comment by skeletonbow** [comment]
> So there IS a channel blur effect!?

Not that I'm aware of.  In my own example screenshots I am using only the StreamFX Blur filter applied to a copy of the chroma keyed source video.  I tested all of the blur types and while they all look slightly different in terms of blur quality they all have the same desired effect for me here.

**Comment by skeletonbow** [comment]
BTW, what operating system, OBS version and StreamFX version are you using?

**Comment by baterax** [comment]
Lemme see. Windows 11, OBS 27.2.4 and...
StreamFX 0.11.1.0-g81a96998

**Comment by Xaymar** [comment]
> So there IS a channel blur effect!?

No. I've repeatedly asked you what you actually wanted to have as applying generic Blur to the Alpha channel results in very bad things - especially in OBS. For example, what color is the background for a fully transparent pixel? White, Black, Gray, Green, Red, Blue, Yellow, Purple, Cyan? This is what Channel Blur would solve as it would act independently on each channel.

**Comment by baterax** [comment]
Yes, I do want to request a Channel Blur. That was the idea initially, but Skeleton showed the current one working in here so I thought it could be done...
Apparently it can't :-(

**Comment by skeletonbow** [comment]
I'm just using the regular blur in OBS 26.1.1 with StreamFX 0.10.0.  I suspect it should work the same in 27.2.4 with 0.11.x but I'll have to set it up there to see if I experience any difference.  One can't rule out some weird OBS bug or issue going on assuming we're doing the same thing.

**Comment by Xaymar** [comment]
From what it looks like, the difference is down to how the Blur is applied. @skeletonbow applies it to a solid color layer/group, which works as expected. @baterax applies it to a layer with transparency, which ends up with transparency being left as it is.

**Comment by skeletonbow** [comment]
My bottom layer is the colorful park image, then I have the media source of Shia LaBeouf on top of that which is an MP4 file with a chroma key filter on it, to simulate a chroma keyed webcam, so this source has a transparent background and shows up over the park image.  Then the source mirror is of the Shia media source so also has transparent background, which I've cropped to the lower portion of the screen, then I put blur filter on it (as an aside, it would be more efficient to use a Crop/Pad filter then blur rather than transform crop, as it'd reduce the amount of pixels fed to blur, but good enough for prototype).  Finally I put a color source with 50% opacity on top and cropped it to the same lower portion of screen.  No other sources or filters, and not using nested scenes nor OBS groups.

Hope this helps.
